File: sdks/python/examples/track_orders.py
"""Subscribe to ChainBridge order events from Python."""

from __future__ import annotations

import asyncio
import os

from chainbridge import AsyncChainBridgeClient
from chainbridge.types import WsEvent


async def on_event(event: WsEvent) -> None:
    print(f"{event.type}: {event.data}")


async def main() -> None:
    async with AsyncChainBridgeClient(api_key=os.environ.get("CHAINBRIDGE_API_KEY")) as client:
        ws = client.create_websocket()
        ws.subscribe("orders", on_event)
        await ws.start()
        try:
            await asyncio.Event().wait()
        finally:
            await ws.close()


if __name__ == "__main__":
    asyncio.run(main())

File: sdks/typescript/examples/track-orders.ts
/**
 * Subscribe to order events on the public WebSocket feed.
 */

import { ChainBridgeClient } from "../src";

async function main() {
  const client = new ChainBridgeClient({
    baseUrl: process.env.CHAINBRIDGE_API_URL ?? "https://api.chainbridge.io",
    apiKey: process.env.CHAINBRIDGE_API_KEY,
  });
  const ws = client.createWebSocket();
  await ws.connect();
  ws.subscribe("orders", (event) => console.log("event:", event.type, event.data));
  console.log("Listening for order events. Ctrl+C to exit.");
}

main();
